plugins/CoreHome/angularjs/common/services/piwik-url.js

```diff
@@ -22,42 +22,11 @@
 
         function getSearchParam(paramName)
         {
-            if (paramName === 'segment') {
-                var hash = window.location.href.split('#');
-                if (hash && hash[1]) {
-                    return piwik.broadcast.getValueFromHash(paramName, hash[1]);
-                }
-
-                return broadcast.getValueFromUrl(paramName);
-            }
-
-            // available in global scope
-            var search = $location.search();
```

Comment:
The code below uses `$location.search()`, and I replaced it with code that uses `broadcast`, since $location is pretty broken. So all of this code which works around bugs in `$location` is no longer needed.